File: 2015/10/solution.py
def look_and_say(nums: list[int]) -> list[int]:
    new_nums = []
    start = 0
    for i in range(len(nums)):
        if nums[i] != nums[start]:
            new_nums.append(i - start)
            new_nums.append(nums[start])
            start = i
    new_nums.append(len(nums) - start)
    new_nums.append(nums[start])
    return new_nums


number = [int(d) for d in input()]
for _ in range(40):
    number = look_and_say(number)
print("Part One:", len(number))
for _ in range(10):
    number = look_and_say(number)
print("Part Two:", len(number))

File: 2015/10/solution.go
package main

import (
	"fmt"
)

func lookAndSay(nums []int) []int {
	var new_nums []int
	start := 0
	for i := range nums {
		if nums[i] != nums[start] {
			new_nums = append(new_nums, i-start, nums[start])
			start = i
		}
	}
	new_nums = append(new_nums, len(nums)-start, nums[start])
	return new_nums
}

func main() {
	var number string
	fmt.Scanln(&number)
	digits := make([]int, len(number))
	for i, d := range number {
		digits[i] = int(d - '0')
	}
	for i := 0; i < 40; i++ {
		digits = lookAndSay(digits)
	}
	fmt.Println("Part One:", len(digits))
	for i := 0; i < 10; i++ {
		digits = lookAndSay(digits)
	}
	fmt.Println("Part Two:", len(digits))
}
